aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -447,13 +476,10 @@ Tensor& prod_out(Tensor& result, const Tensor& self, Dimname dim,
 
 Tensor &mean_out_cpu_gpu(Tensor &result, const Tensor &self, IntArrayRef dim,
                  bool keepdim, c10::optional<ScalarType> opt_dtype) {
-  ScalarType scalarType = opt_dtype.has_value() ? opt_dtype.value() : self.scalar_type();
-  TORCH_CHECK(
-      at::isFloatingType(scalarType) || at::isComplexType(scalarType),
-      "Can only calculate the mean of floating types. Got ",
-      toString(scalarType),
-      " instead.");
-  ScalarType dtype = get_dtype(result, self, opt_dtype, true);
+  ScalarType dtype = get_dtype(result, self, opt_dtype);
+  TORCH_CHECK(!at::isIntegralType(dtype),
```

Comment:
pass includeBool?

Reply:
Correct. And a good argument against default args. The tests also need to be expanded to include bool, int, and float tests cases.